### PR TITLE
DOC-4216

### PR DESCRIPTION
**Pull request by jamesnocentini**
https://issues.couchbase.com/browse/DOC-4216

### CODE REVIEW

File: modules/ROOT/pages/data-routing.adoc [hunk 1]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+* The value of revs_limit.
+Indeed, the sync metadata is retained for each non-leaf revision.
+
+The following table estimates the allowed channel count per document according to different parameters when shared bucket access is disabled/enabled.
+
+|===
+|`enable_shared_bucket_access: false`|`enable_shared_bucket_access: true`
+
+2+a|*Sync metadata size limit*
+|20 Mb per document
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|20 MB per document
```

File: modules/ROOT/pages/data-routing.adoc [hunk 2]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+|1 Mb per document
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|1 MB per document
```

File: modules/ROOT/pages/data-routing.adoc [hunk 3]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+
+2+a|*Formula to find the max channel count per document*
+|(20Mb - doc size) / (average channel name size * revs_limit)
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|(20MB - doc size) / (average channel name size * revs_limit)
```

File: modules/ROOT/pages/data-routing.adoc [hunk 4]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+|1Mb / (average channel name size * revs_limit)
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|1MB / (average channel name size * revs_limit)
```

File: modules/ROOT/pages/data-routing.adoc [hunk 5]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+2+a|*Example 1: doc size = 5 Mb, average channel name size = 25 bytes, revs_limit = 100*
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
2+a|*Example 1: doc size = 5 MB, average channel name size = 25 bytes, revs_limit = 100*
```

File: modules/ROOT/pages/data-routing.adoc [hunk 6]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+|(20Mb - 5Mb)/(25bytes * 100) = 6000
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|(20MB - 5MB)/(25 bytes * 100) = 6000
```

File: modules/ROOT/pages/data-routing.adoc [hunk 7]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+|1Mb/(25 bytes * 100) = 400
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|1MB/(25 bytes * 100) = 400
```

File: modules/ROOT/pages/data-routing.adoc [hunk 8]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+2+a|*Example 1: doc size = 5 Mb, average channel name size = 25 bytes, revs_limit = 1000*
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
2+a|*Example 1: doc size = 5 MB, average channel name size = 25 bytes, revs_limit = 1000*
```

File: modules/ROOT/pages/data-routing.adoc [hunk 9]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+|(20Mb - 5Mb)/(25bytes * 1000) = 600
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|(20MB - 5MB)/(25 bytes * 1000) = 600
```

File: modules/ROOT/pages/data-routing.adoc [hunk 10]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+|1Mb/(25 bytes * 1000) = 40
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
|1MB/(25 bytes * 1000) = 40
```

File: modules/ROOT/pages/data-routing.adoc [hunk 11]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+
+The sync metadata contains the list of channel names to which a document is mapped.
+When a document is assigned to a new channel, the channel name is appended to that document's sync metadata.
+
+Therefore, the size that is occupied by channel sync metadata varies based on the following.
+
+* The channel name.
+A shorter channel name will occupy less space ("customer==0030169303" vs "cs==0030169303").
+* The value of revs_limit.
+Indeed, the sync metadata is retained for each non-leaf revision.
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
Indeed, a copy of channel metadata is retained for each revision of a document.
```

We only have one sync metadata per document, but inside that there is one copy of channel metadata per revision.

File: modules/ROOT/pages/data-routing.adoc [hunk 12]
```diff
@@ -50,6 +50,40 @@ Creating a xref:sync-function-api.adoc[sync function] is a more flexible way to
...
+
+The following table estimates the allowed channel count per document according to different parameters when shared bucket access is disabled/enabled.
```
Comment (bbrks) [inline modules/ROOT/pages/data-routing.adoc]:
```suggestion
The following table estimates the channel count limits per document according to different parameters when shared bucket access is disabled/enabled.
```

**Review by bbrks — CHANGES_REQUESTED** [review]
I've tweaked some of the wording, and changed the units `Mb` -> `MB`.

I think GitHub lets you accept these suggestions in bulk through the web UI now if you agree.

**Comment by jamesnocentini** [comment]
@bbrks great thanks.
And wow the bulk accept suggestion feature is so good.

**Comment by jamesnocentini** [comment]
@adamcfraser, awaiting your PR approval (for correctness).

@bbrks can you OK the PR following the changes?

**Comment by bbrks** [comment]
Hey @jamiltz ,

I've discussed this change with @adamcfraser and I think we're going to change tack a little bit.

We'll be improving the stats monitoring and error/warnings related to this inside Sync Gateway for the next version, and instead provide a section in the docs with a "What to do when your channel count nears/exceeds usable space" so the user can solve their problem.

The new section should describe that in order to alleviate oversized channel metadata, they should lower one or more of the following:
1. number of channels per document
2. channel name lengths
3. revs_limit

**Comment by jamesnocentini** [comment]
Thanks @bbrks , updated the PR. Let me know if it needs further modifications.

**Review by bbrks — APPROVED** [review]
Cool, thanks for those changes @jamiltz

**Pull request merged**